Client/Frontend/Browser/LoginsHelper.swift

```diff
@@ -142,36 +142,31 @@ class LoginsHelper: TabHelper {
             return
         }
 
-        let promptMessage: NSAttributedString
+        let promptMessage: String
         if let username = login.username {
-            let promptStringFormat = NSLocalizedString("LoginsHelper.PromptSaveLogin.Title", value: "Save login %@ for %@?", comment: "Prompt for saving a login. The first parameter is the username being saved. The second parameter is the hostname of the site.")
-            promptMessage = NSAttributedString(string: String(format: promptStringFormat, username, login.hostname))
+            promptMessage = String(format: Strings.SaveLoginUsernamePrompt, username, login.hostname)
         } else {
-            let promptStringFormat = NSLocalizedString("LoginsHelper.PromptSavePassword.Title", value: "Save password for %@?", comment: "Prompt for saving a password with no username. The parameter is the hostname of the site.")
-            promptMessage = NSAttributedString(string: String(format: promptStringFormat, login.hostname))
+            promptMessage = String(format: Strings.SaveLoginPrompt, login.hostname)
         }
 
-        if snackBar != nil {
-            tab?.removeSnackbar(snackBar!)
+        if let existingPrompt = self.snackBar {
+            tab?.removeSnackbar(existingPrompt)
         }
 
-        snackBar = TimerSnackBar(attrText: promptMessage,
```

Comment:
Rewrote the SnackBar API here. Made it follow UIAlertView more. This avoids all the nesting